src/azure-cli/azure/cli/command_modules/profile/_help.py

```diff
@@ -10,42 +10,50 @@
-    - name: Log in with a service principal using client certificate.
-      text: az login --service-principal -u http://azure-cli-2016-08-05-14-31-15 -p ~/mycertfile.pem --tenant contoso.onmicrosoft.com
-    - name: Log in using a VM's system-assigned managed identity.
-      text: az login --identity
-    - name: Log in using a VM's user-assigned managed identity. Client or object ids of the service identity also work.
-      text: az login --identity -u /subscriptions/<subscriptionId>/resourcegroups/myRG/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myID
+- name: Log in interactively.
+  text: az login
+- name: Log in with user name and password. This doesn't work with Microsoft accounts or accounts that have two-factor authentication enabled.
+  text: az login --user johndoe@contoso.com --password VerySecret
+- name: Log in with a password containing a `-` as the first character.
+  text: az login --user johndoe@contoso.com --password=-myPasswordThatStartsWithAdash
+- name: Log in with a service principal using a client secret.
+  text: az login --service-principal --user http://azure-cli-2016-08-05-14-31-15 --password VerySecret --tenant contoso.onmicrosoft.com
+- name: Log in with a service principal using a client certificate.
+  text: aaz login --service-principal --user http://azure-cli-2016-08-05-14-31-15 --password ~/MyCertfile.pem --tenant contoso.onmicrosoft.com
+- name: Log in using a VM's system assigned identity.
+  text: az login --identity
+- name: Log in using a VM's user assigned identity. Client or object ids of the service identity also work.
+  text: az login --identity --user /subscriptions/MySubscriptionID/resourcegroups/MyResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myID
 """
 
 helps['account'] = """
 type: group
 short-summary: Manage Azure subscription information.
+long-summary: >
+    Use `az login` to log into Azure before managing subscriptions.
+    For an in-depth overview of working with subscriptions using the Azure CLI, see
+    [How to manage Azure subscriptions with the Azure CLI](https://learn.microsoft.com/cli/azure/manage-azure-subscriptions-azure-cli).
 """
 
 helps['account clear'] = """
 type: command
 short-summary: Clear all subscriptions from the CLI's local cache.
-long-summary: To clear the current subscription, use 'az logout'.
+long-summary: To clear the current subscription, use `az logout`.
 """
 
 helps['account list'] = """
 type: command
-short-summary: >-
-    Get a list of subscriptions for the logged in account. By default, only 'Enabled' subscriptions from the current
-    cloud is shown.
+short-summary: Get a list of subscriptions for the logged in account.
+examples:
+- name: List all subscriptions.
+  text: az account list --all
+- name: List all enabled subscriptions.
+  text: az account list
+- name: Get the current default subscription.
+  text: az account list --query "[?isDefault]"
```

Comment:
As discussed in https://github.com/Azure/azure-cli/pull/24332#discussion_r1003912453, I am hesitating on whether we want to add examples for generic argument `--query`.